pandas/core/indexing.py

```diff
@@ -1091,8 +1095,17 @@ def _convert_to_indexer(self, obj, axis=0, is_setter=False):
         """
         labels = self.obj._get_axis(axis)
 
-        # if we are a scalar indexer and not type correct raise
-        obj = self._convert_scalar_indexer(obj, axis)
+        if isinstance(obj, slice):
+            return self._convert_slice_indexer(obj, axis)
+
+        # try to find out correct indexer, if not type correct raise
+        try:
+            obj = self._convert_scalar_indexer(obj, axis)
+        except TypeError:
+
+            # but we will allow setting
+            if is_setter:
```

Comment:
Hmm, this seems great (on your branch)

```python
In [10]: s = Series([1,2,3])

In [11]: s[1.] = 5

In [12]: s
Out[12]:
0    1
1    2
2    3
1    5
dtype: int64

In [13]: s.index
Out[13]: Int64Index([0, 1, 2, 1], dtype='int64')
```

So before this would have warned about the Floating scalar, cast to an int, and then overwritten the value at `index=1`. I guess that would make this an API change, but this seems like a pretty hairy edge-case

I think I see why you would want to allow setting of floats that could be cast as integers, but maybe you could explain your thinking a bit more?